src/action_cc.cc

```diff
@@ -552,6 +556,168 @@ NM_ACTION_(nickel_wifi) {
+    // Device::hasOrientationSensor, not the function itself.
+    //
+    // If the device has an orientation sensor, the user can optionally set the
+    // [DeveloperSettings] ForceAllowLandscape setting to true or apply the
+    // "Allow rotation on all devices" patch to make the rotation take effect
+    // even if the current view doesn't normally allow it.
+    //
+    // This is because in addition to the previous limitation, if
+    // [DeveloperSettings] ForceAllowLandscape if not true, the orientation is
+    // limited based on if Device::hasOrientationSensor and the allowed
+    // orientations for the current view.
+
+    NM_LOG(Device_hasOrientationSensor(dev)
+        ? "nickel_orientation: if this didn't do anything, you may need to set [DeveloperSettings] ForceAllowLandscape=true to allow landscape orientations on all views"
+        : "nickel_orientation: if this didn't do anything, ensure you have applied the 'Allow rotation on all devices' kobopatch patch, or that you have set [DeveloperSettings] ForceAllowLandscape=true");
+
+    // Set the orientation.
+    //
+    // This is the only thing which is really required here, but if used alone,
+    // the orientation sensor may override it, and the orientation will only be
+    // set if it is within the constraints of the current rotation mode of
+    // auto/portrait/landscape.
+    //
+    // Note that this function is actually part of the private Qt QPA API, and
+    // Nickel really should be doing this part of the rotation logic (including
+    // the sensors) from the libkobo QPA plugin. But, this is how Nickel does
+    // it, so we need to do it the same way (this does make it easier for both
+    // us and Kobo, I guess, due to how intertwined the orientation stuff is).
+    //
+    // This is the same function which is called by Nickel's handler for the
+    // orientation sensor's reading.
+
+    QWindowSystemInterface_handleScreenOrientationChange(QGuiApplication::primaryScreen(), o);
+
+    // We don't update the status bar rotation icon, since we would need to get
+    // ahold of the StatusBarView, and I don't like the current options for
+    // doing that in a version-agnostic way: walking the entire QObject tree,
+    // hooking the constructor, or using offsets directly to get ahold of the
+    // StatusBarView from the StatusBarController from the MainWindowController.
+    // I might reconsider this in the future if there is demand for it.
+
+    if ((icon_mask&o) == 0)
+        NM_LOG("nickel_orientation: the status bar rotate icon may be out of date (this is expected)");
+
+    // If we ever wanted to add an option for setting to rotation to auto/locked
+    // portrait/locked landscape, we'd need to call RotatePopupController
+    // ::on{Auto,Portrait,Landscape} on StatusBarView::rotatePopupController.
+    // This is how the rotate popup and the dropdown in the reading settings
+    // does it.
```

Comment:
I've improved this to hide most of the implementation details (around that this uses the same method as the orientation sensor reading handler). This action will now change the rotation regardless of the current locked/auto rotation setting. In addition, it should override the orientation sensor (can you confirm this, @NiLuJe?) until a reboot or until the rotation is changed from the normal drop-down.

For devices without a rotation sensor, `ForceAllowLandscape` is needed for the MainWindowController to accept the rotation. Optionally, the `Allow rotation on all devices` patch can be used to also show the built-in rotation menu.

For devices with a sensor, `ForceAllowLandscape` can be optionally used to allow the rotation to take effect even on views which don't usually allow landscape.

The only limitation of this is that the status bar rotation icon won't be updated (it's possible to do, but I don't think it's worth the performance/safety/firmware versioning cost).

Reply:
I'll test on a gyro device tomorrow if no-one beats me to it ;).

Reply:
Thanks! I've left some notes about what exactly needs testing in my [comment](https://github.com/pgaskin/NickelMenu/pull/70#pullrequestreview-461310625) at the bottom. In short, I'm mainly concerned about ensuring the transition between sensor/no sensor works properly, and that I didn't miss any spots I need to override the sensor's value.

Reply:
@NiLuJe, one other thing: do you know of any devices where `ForceAllowLandscape` doesn't affect the touchscreen properly?

Reply:
I've never played with it myself, but it's *quite* possible that *at least* the touch translation could be off in some circumstances (i.e., old FW, old devices).

Reply:
Which device did you do your testing on?

Reply:
@NiLuJe responded below: [a forma](https://github.com/pgaskin/NickelMenu/pull/70#issuecomment-669514661).

Reply:
> but it's quite possible that at least the touch translation could be off in some circumstances (i.e., old FW, old devices).

I guess I'll just leave it for now and see if anyone complains. Even in those old posts I mentioned, the issues weren't always reproducible. Anyways, if something like that does happen with this action, it will only take a reboot to put it back to normal.